projection/core/src/main/scala/com/lightbend/lagom/projection/State.scala

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (C) 2016-2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package com.lightbend.lagom.projection
+
+import akka.annotation.ApiMayChange
+import com.lightbend.lagom.internal.projection.ProjectionRegistryActor.WorkerCoordinates
+
+import scala.util.control.NoStackTrace
+
+@ApiMayChange
+// TODO: provide serialisers (all the ADT is on LWWMaps so must be serializable)
+sealed trait Status
+sealed trait Stopped extends Status
+case object Stopped  extends Stopped
+sealed trait Started extends Status
+case object Started  extends Started
+
+@ApiMayChange
+case class Worker(tagName: String, key: String, requestedStatus: Status, observedStatus: Status)
+
+@ApiMayChange
+case class Projection(name: String, workers: Seq[Worker])
+
+@ApiMayChange
+case class State(projections: Seq[Projection]) {
+  def findProjection(projectionName: String): Option[Projection] =
+    projections.find(_.name == projectionName)
+
+  def findWorker(workerKey: String): Option[Worker] =
+    projections.flatMap(_.workers).find(_.key == workerKey)
+}
+
+object State {
+
+  type WorkerKey = String
+  private[lagom] def fromReplicatedData(
+      nameIndex: Map[WorkerKey, WorkerCoordinates],
+      requestedStatusLocalCopy: Map[WorkerKey, Status],
+      observedStatusLocalCopy: Map[WorkerKey, Status]
+  )(defaultRequested: Status, defaultObserved: Status = Stopped): State = {
+
+    val workers: Map[String, Seq[Worker]] = nameIndex
+      .map {
+        case (workerKey, coordinates) =>
+          val w = Worker(
+            coordinates.tagName,
+            coordinates.asKey,
+            requestedStatusLocalCopy.getOrElse(workerKey, defaultRequested),
+            observedStatusLocalCopy.getOrElse(workerKey, defaultObserved)
+          )
+          w -> coordinates.projectionName
+      }
+      .toMap
```

Comment:
Looks like this `toMap` and the one a few lines below are not necessary.

Reply:
You are 50% right. The first one is unnecessary but the second one is to calm the [`2.13` Gods](https://github.com/lagom/lagom/pull/2055/commits/57297e0ebcf34d6f93e43f53e95639aa0db2e410):

```

[info] Compiling 1 Scala source to /Users/ignasi/git/projects/lightbend/lagom/lagom/projection/core/target/scala-2.13/classes ...
[error] /Users/ignasi/git/projects/lightbend/lagom/lagom/projection/core/src/main/scala/com/lightbend/lagom/projection/State.scala:58:17: type mismatch;
[error]  found   : scala.collection.MapView[String,Seq[com.lightbend.lagom.projection.Worker]]
[error]  required: Map[String,Seq[com.lightbend.lagom.projection.Worker]]
[error]       .mapValues(_.keys.toSeq)
[error]                 ^
[error] one error found
[error] (Compile / compileIncremental) Compilation failed

```